src/date-picker/date-picker.tsx

```diff
@@ -0,0 +1,1311 @@
+
+        const isStartBeforeRange = dayjs(start).isBefore(range);
+        const isRangeAfterStart = dayjs(range).isAfter(start);
+
+        return {
+            isStartBeforeRange,
+            isRangeAfterStart,
+            start,
+            range,
+        };
+    };
+
+    const generateDayStatus = (day: Date): GenerateDayStatusValue => {
+        const _dateStartWithDay = new Intl.DateTimeFormat("en-GB", {
+            day: "2-digit",
+            month: "2-digit",
+            year: "numeric",
+        })
+            .format(day)
+            .replaceAll("/", "-");
+
+        // YYYY-MM-DD
+        const dateStartWithYear = revertedDateFormat(_dateStartWithDay);
+
+        let isDisabled = false;
+        if (disabledDates && disabledDates.length) {
+            isDisabled = disabledDates.includes(dateStartWithYear);
+        }
+
+        let disabledBefore = false;
+        let disabledAfter = false;
+
+        const { start, range, startStatus, rangeStatus } =
+            controlTransitionValue();
+
+        const variant: VariantDay =
+            calendarDate?.clone().toDate().getMonth() !== day.getMonth()
+                ? "nextMonth"
+                : dayjs(start).isSame(day, "day")
+                ? "startSelected"
+                : dayjs(range).isSame(day, "day")
+                ? "rangeSelected"
+                : dayjs().isSame(day, "day")
+                ? "today"
+                : "default";
+
+        if (type === "range") {
+            disabledBefore = dayjs(day).isBefore(start, "day");
+            disabledAfter = dayjs(day).isAfter(range, "day");
+
+            if (range.length || start.length) {
+                if (focusTo.container === "range") {
+                    disabledAfter = false;
+                }
+
+                if (focusTo.container === "start") {
+                    disabledBefore = false;
+                }
+            }
+
+            if (startStatus === "confirmed" && rangeStatus === "confirmed") {
+                // remove disabled if alredy been confirmed before
+                disabledBefore = false;
+                disabledAfter = false;
+            }
+        }
+
+        return {
+            dateStartWithYear,
+            isDisabled,
+            variant,
+            disabledBefore,
+            disabledAfter,
+        };
+    };
+
+    const generateDayClass = (date: string, variant: VariantDay): string => {
+        const classes: string[] = [];
+        const { start, range, startStatus, rangeStatus } =
+            controlTransitionValue();
+
+        const rangeClasses = handleSelectRange(date);
+        const startClasses = handleSelectStart(date);
+        const hoverClasses = handleHovering(date);
+
+        /* handle selected between date */
+        if (
+            start.length &&
+            range.length &&
+            dayjs(date).isBetween(start, range, "day", "[]")
+        ) {
+            classes.push("selected-between pre-selected-between");
```

Comment:
I'm not comfortable about interchanging usage of class names for styled components. Would usage of props not suffice?